File: packages/schemas/src/sourceDocument.ts
import { z } from "zod";
import { Id, IsoDateTime, Url } from "./primitives.js";

export const SourceDocument = z.object({
  id: Id,
  url: Url,
  platform: z.string().min(1),
  author: z.string().nullable(),
  capturedAt: IsoDateTime,
  publishedAt: IsoDateTime.nullable(),
  content: z.string().optional(),
  contentHash: z.string().optional(),
  context: z.string().optional(),
});
export type SourceDocument = z.infer<typeof SourceDocument>;

File: backend/src/main/java/com/acltabontabon/kuro/domain/SourceDocument.java
package com.acltabontabon.kuro.domain;

import java.time.OffsetDateTime;

// mirrors packages/schemas/src/sourceDocument.ts; author/publishedAt are nullable
// by design (PII posture), content/contentHash/context are optional
public record SourceDocument(
        String id,
        String url,
        String platform,
        String author,
        OffsetDateTime capturedAt,
        OffsetDateTime publishedAt,
        String content,
        String contentHash,
        String context) {
}
